client/signup/site-mockup/index.jsx

```diff
@@ -137,53 +127,86 @@ class SiteMockups extends Component {
-			'is-font-loading': ! this.state.fontLoaded,
-			'is-font-error': ! this.state.fontError,
 		} );
+		const langSlug = getLocaleSlug();
+		const language = getLanguage( langSlug );
+		const isRtl = language && language.rtl;
 		const otherProps = {
-			title,
-			tagline: this.getTagline(),
-			content: this.getContent( verticalPreviewContent ),
-			siteType,
-			siteStyle,
+			fontUrl,
+			cssUrl: getThemeCssUri( themeSlug, isRtl ),
+			content: {
+				title,
+				tagline: this.getTagline(),
+				body: this.getContent( verticalPreviewContent ),
+			},
+			langSlug,
+			isRtl,
+			onPreviewClick: this.handleClick,
+			className: siteStyle,
 		};
-		const fontStyle = getCSS( `.site-mockup__content`, siteStyle, siteType );
 
 		return (
 			<div className={ siteMockupClasses }>
-				{ ! this.state.fontError && <style>{ fontStyle }</style> }
-
-				<div className="site-mockup__help-tip">
-					<p>
-						{ translate(
-							'Scroll down to see your website. Once you complete setup you’ll be able to customize it further.'
-						) }
-					</p>
-					<Gridicon icon="chevron-down" />
-				</div>
-
+				{ shouldShowHelpTip && <SiteMockupHelpTip /> }
 				<div className="site-mockup__devices">
-					<SiteMockup size="desktop" { ...otherProps } />
-					<SiteMockup size="mobile" { ...otherProps } />
+					<SignupSitePreview
+						defaultViewportDevice="desktop"
+						resize={ true }
+						scrolling={ false }
+						{ ...otherProps }
+					/>
+					<SignupSitePreview defaultViewportDevice="phone" { ...otherProps } />
 				</div>
 			</div>
 		);
 	}
 }
 
-export default connect( state => {
-	const siteInformation = getSiteInformation( state );
-	return {
-		title: siteInformation.title || translate( 'Your New Website' ),
-		address: siteInformation.address,
-		phone: siteInformation.phone,
-		siteStyle: getSiteStyle( state ),
-		siteType: getSiteType( state ),
-		verticalPreviewContent: getSiteVerticalPreview( state ),
-	};
-} )( SiteMockups );
+export default connect(
+	( state, ownProps ) => {
+		const siteInformation = getSiteInformation( state );
+		const siteStyle = getSiteStyle( state );
+		const siteType = getSiteType( state );
+		const styleOptions = getSiteStyleOptions( siteType );
+		const style = find( styleOptions, { id: siteStyle || 'professional' } );
+		return {
+			title: siteInformation.title || translate( 'Your New Website' ),
+			address: siteInformation.address,
+			phone: siteInformation.phone,
+			siteStyle,
+			siteType,
+			verticalPreviewContent: getSiteVerticalPreview( state ),
+			verticalSlug: getSiteVerticalSlug( state ),
+			shouldShowHelpTip:
+				'site-topic-with-preview' === ownProps.stepName ||
+				'site-information-title-with-preview' === ownProps.stepName,
+			themeSlug: style.theme,
+			fontUrl: style.fontUrl,
+		};
+	},
+	dispatch => ( {
+		handleClick: ( verticalSlug, siteStyle, size ) =>
+			dispatch(
+				recordTracksEvent( 'calypso_signup_site_preview_mockup_clicked', {
```

Comment:
I don't think this event is recorded - is the click handler getting attached properly?

Reply:
It's triggered by a click on the iframe body. 

Works for me:

![Apr-18-2019 07-40-05](https://user-images.githubusercontent.com/6458278/56322882-96474380-61ad-11e9-903e-ca5a818b1a82.gif)

Any way I can reproduce what you're experiencing?

I can fix directly after merge if it persists.

Reply:
All good, I think it was just my ublock :-)

Reply:
Block all that WordPress spam! Hehe